### PR TITLE
Added custom layout to the spinner adapter in the card browser activity to prevent the overlapping of various interactive components on the screen

### PR DESCRIPTION
**Pull request by jaivsh**
## Purpose / Description
In the card browser, when the spinner 2 is clicked, it expands to the whole of the screen height wise, which is causing the overlap of the spinner items on top of UI elements on the upper action bar. 

## Fixes
I've rescaled the layout of the spinner options so as  

## Approach
I've created a custom layout to be used as the drop down view resource of the spinner adapters. In that layout, I've rescaled the sizes of spinner options by observing its behavior on various different testing devices, so as to fix the overlap problem and make the user experience better. 

## How Has This Been Tested?

I've run and verified the implementation on pixel 2(1080x1920) mobile, pixel 3a(1080x2220) mobile, and nexus 7(1220x1920) tablet emulators. Apart from that, I've also tested it on Motorola zoom tablet and Redmi note 3 physical devices to verify the new design and behavior.

## Checklist

- [Y] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [Y] You have a descriptive commit message with a short title (first line, max 50 chars).
- [Y] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [Y] You have commented your code, particularly in hard-to-understand areas
- [Y] You have performed a self-review of your own code
- [Y] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [Y] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Comment by mikehardy** [comment]
It appears to have a tiny lint error:

```

> Task :AnkiDroid:ktlintMainSourceSetCheck FAILED
/home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt:583:9 Missing space after //

```

I'll still flag it for review though, missing one space character must be fixed but should not block review I think

**Comment by jaivsh** [comment]
Hello, @mikehardy, I've updated all the things here. Thanks for telling this!

File: AnkiDroid/src/main/res/values/styles.xml [hunk 1]
```diff
@@ -150,9 +150,13 @@
...
     </style>
 
+    <style name="card_browser_spinner_child_height">
+        <item name="android:layout_height">36dp</item>
+    </style>
+
 
     <style name="reviewer_whiteboard_editor_button_style">
-        <item name="android:layout_height">40dp</item>
+        <item name="android:layout_height">40dp</item>ac
```
Comment (mikehardy) [inline AnkiDroid/src/main/res/values/styles.xml:159]: Looks like a typo in there ?

```suggestion
        <item name="android:layout_height">40dp</item>
```

File: AnkiDroid/src/main/res/values/styles.xml [hunk 2]
```diff
@@ -117,7 +117,7 @@
 
     <!-- For all other action bar popups like overflow menu (except spinner dropdown in Lollipop). -->
     <style name="ActionBar.Popup" parent="Base.ThemeOverlay.AppCompat.ActionBar">
-        <item name="android:drawSelectorOnTop">true</item>
+        <item name="android:drawSelectorOnTop">false</item>
```
Comment (mikehardy) [inline AnkiDroid/src/main/res/values/styles.xml:120]: will this affect other selectors? I don't think so but we have deck spinners in lots of dialogs, your testing on this specific change was very thorough (thank you!) and I just want to know if you also looked at note edit screen deck selection, card-browser long-press multi-select mode change deck spinner

**Review by mikehardy — CHANGES_REQUESTED** [review]
Thorough testing - fantastic, one nit typo, and one question on deck spinner interaction with the change to action bar draw on top boolean

**Comment by mikehardy** [comment]
By the way - really sorry for the delay, it's taken me this long to get back to just the "review" queue from the "pending merge" and "second review" queues - really hectic the last couple weeks!

**Review by mikehardy — APPROVED** [review]
comments appear resolved

**Pull request merged**